addOns/scripts/src/main/java/org/zaproxy/zap/extension/scripts/ConsolePanel.java

```diff
@@ -481,7 +488,11 @@ private void updateCommandPanelState(ScriptWrapper script) {
         getCommandPanel().clear();
         getCommandPanel().setCommandScript(script.getContents());
         getCommandPanel().setScriptType(script.getTypeName());
-        getCommandPanel().setCommandCursorPosition(0);
+        if (this.scriptWrapperToOffset.containsKey(script)) {
+            getCommandPanel().setCommandCursorPosition(this.scriptWrapperToOffset.get(script));
```

Comment:
`setCommandCursorPosition` needs to be changed to not set the position to more than the length of the script (e.g. if the script is changed outside ZAP and has now less chars than the previous position).